packages/@react-spectrum/tree/src/TreeView.tsx

```diff
@@ -0,0 +1,361 @@
+  );
+}
+
+interface TreeRowRenderProps extends TreeItemRenderProps {
+  isLink?: boolean
+}
+
+const treeRow = style<TreeRowRenderProps>({
+  position: 'relative',
+  display: 'flex',
+  height: 10,
+  width: 'full',
+  boxSizing: 'border-box',
+  fontSize: 'base',
+  fontWeight: 'normal',
+  lineHeight: 200,
+  color: 'body',
+  outlineStyle: 'none',
+  cursor: {
+    default: 'default',
+    isLink: 'pointer'
+  },
+  // TODO: not sure where to get the equivalent colors here, for instance isHovered is spectrum 600 with 10% opacity but I don't think that exists in the theme
+  backgroundColor: {
+    isHovered: '[var(--spectrum-table-row-background-color-hover)]',
+    isFocusVisibleWithin: '[var(--spectrum-table-row-background-color-hover)]',
+    isPressed: '[var(--spectrum-table-row-background-color-down)]',
+    isSelected: '[var(--spectrum-table-row-background-color-selected)]'
+  }
+});
+
+const treeCellGrid = style({
+  display: 'grid',
+  width: 'full',
+  alignItems: 'center',
+  // TODO: needed to use spectrum var since gridTemplateColumns uses baseSizing and not scaled sizing
+  gridTemplateColumns: ['minmax(0, auto)', 'minmax(0, auto)', 'minmax(0, auto)', 'var(--spectrum-global-dimension-size-500)', 'minmax(0, auto)', '1fr', 'minmax(0, auto)', 'auto'],
+  gridTemplateRows: '1fr',
+  gridTemplateAreas: [
+    'drag-handle checkbox level-padding expand-button icon content actions actionmenu'
+  ],
+  color: {
+    isDisabled: {
+      default: 'gray-400',
+      forcedColors: 'GrayText'
+    }
+  }
+});
+
+// TODO: These styles lose against the spectrum class names, so I've did unsafe for the ones that get overridden
+const treeCheckbox = style({
+  gridArea: 'checkbox',
+  transitionDuration: '160ms',
+  marginStart: 3,
+  marginEnd: 0
+});
+
+const treeIcon = style({
+  gridArea: 'icon',
+  marginEnd: 2
+});
+
+const treeContent = style<Pick<TreeItemContentRenderProps, 'isDisabled'>>({
+  gridArea: 'content',
+  textOverflow: 'ellipsis',
+  whiteSpace: 'nowrap',
+  overflow: 'hidden'
+});
+
+const treeActions = style({
+  gridArea: 'actions',
+  flexGrow: 0,
+  flexShrink: 0,
+  /* TODO: I made this one up, confirm desired behavior. These paddings are to make sure the action group has enough padding for the focus ring */
+  marginStart: .5,
+  marginEnd: 1
+});
+
+const treeActionMenu = style({
+  gridArea: 'actionmenu',
+  width: 8
+});
+
+const treeRowOutline = style({
+  content: '',
+  display: 'block',
+  position: 'absolute',
+  insetStart: 0,
+  insetEnd: 0,
+  top: {
+    default: 0,
+    isFocusVisible: '[-2px]',
+    isSelected: {
+      default: '[-1px]',
+      isFocusVisible: '[-2px]'
+    }
+  },
+  bottom: 0,
+  pointerEvents: 'none',
+  forcedColorAdjust: 'none',
+
+  boxShadow: {
+    isFocusVisible: '[inset 2px 0 0 0 var(--spectrum-alias-focus-color), inset -2px 0 0 0 var(--spectrum-alias-focus-color), inset 0 -2px 0 0 var(--spectrum-alias-focus-color), inset 0 2px 0 0 var(--spectrum-alias-focus-color)]',
```

Comment:
probably could use outline for the focus ring like we are in s2. might simplify this a bit.

Reply:
Sure, added it to the followup ticket to try